oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/directory/IndexSanityChecker.java

```diff
@@ -118,4 +122,54 @@ private static void deleteAllFiles(Directory dir) throws IOException {
             dir.deleteFile(fileName);
         }
     }
+    
+    
+    /**
+     * Accumulated statistics across multiple invocations of the IndexSanityChecker
+     *
+     */
+    public static class IndexSanityStatistics {
+        
+        long totalDurationInMs;
+        long totalIndexSize;
+        
+        /**
+         * Record the time spend to check the sanity of indexes
+         * @param milis the time in miliseconds
+         */
+        public void addDuration(long milis) {
+            totalDurationInMs += milis;
+        }
+        
+        /**
+         * Return the accumulated time for checking the sanity of indexes
+         * @return the total duration in miliseconds
+         */
+        public long getAccumulatedDuration() {
+            return totalDurationInMs;
+        }
+        
+        /**
+         * Record the size of an index considered
+         * @param bytes
+         */
+        public void addIndexSize (long bytes) {
+            totalIndexSize += bytes;
+        }
+        
+        /**
+         * Return the accumulated index size 
+         * @return index size in bytes
+         */
+        public long getAccumulatedIndexSize() {
+            return totalIndexSize;
+        }
+        
+        @Override
+        public String toString() {
+            return String.format("[duration: %d ms, "
+                    + "checked index size: %d bytes (%s)]", totalDurationInMs,
+                    totalIndexSize,IOUtils.humanReadableByteCount(totalIndexSize));
```

Comment:
minor: missing space
```suggestion
                    totalIndexSize, IOUtils.humanReadableByteCount(totalIndexSize));
```